client/state/posts/utils.js

```diff
@@ -91,3 +98,60 @@ export function mergeIgnoringArrays( object, ...sources ) {
 		}
 	} );
 }
+
+/**
+ * Given a post object, returns a normalized post object
+ *
+ * @param  {Ojbect} post Raw edited post object
+ * @return {Object}      Normalized post object
+ */
+export function normalizePost( post ) {
+	if ( ! post ) {
+		return null;
+	}
+
+	return normalizeFlow( post );
+}
+
+/**
+ * Takes existing term post edits and updates the `terms_by_id` attribute
+ *
+ * @param  {Object}    post  object of post edits
+ * @return {Object}          normalized post edits
+ */
+export function getTermIdsFromEdits( post ) {
+	if ( ! post || ! post.terms ) {
+		return post;
+	}
+	const normalizedPost = cloneDeep( post );
+
+	// Skip "default" taxonomies until legacy token-field and category-selector are removed
+	let taxonomies = omit( post.terms, [ 'post_tag', 'category' ] );
+
+	// Filter taxonomies that are set as arrays ( i.e. tags )
+	// This can be detected by an array of strings vs an array of objects
+	taxonomies = reduce( taxonomies, ( prev, taxonomyTerms, taxonomyName ) => {
+		// Ensures we are working with an array
+		const termsArray = map( taxonomyTerms );
```

Comment:
>It is to ensure we are working with an array since the API returns an object for the post `terms`, toArray should work

Yeah, I seem to recall you mentioning it was something to do with coercing into an array. I think `toArray` would be more clear to understand vs. `map`.